e2e-cypress/files/support/test-evidence.ts

```diff
@@ -65,6 +65,10 @@ export const takeScreenshotEvidence = (testName: string, testStep: number, testS
         logEvidence(testName, testStep, description, [
           `Stored screenshot "${path.basename(result.path)}" with hash (sha256) ${result.hash} taken at ${String(data.takenAt)} as evidence.`
         ]);
+
+        // Create a relative path from the screenshots folder to the mochawesome test-results folder
```

Comment:
What exactly is happening here? A comment for that would be good. Also, is this a recursion with the risk of having an endless loop?

Reply:
The addScreenshot function is for adding an screenshot as context to the html reporter (it is defined in e2e.ts), it's not the same function as the takeScreenshotEvidence function, it's called at the end of it so that all screenshots taken in this way appear in the html report that's later converted to a pdf. Is this what you meant?

Reply:
Ah, now I got it. Thanks for clarifying :)